relayer/src/message_relayer/eth_to_gear/paid_token_transfers.rs

```diff
@@ -81,6 +87,22 @@ impl Relayer {
             genesis_time,
         );
 
+        let client = api_provider
+            .gclient_client(&suri)
+            .expect("failed to create gclient");
+        let remoting = GClientRemoting::new(client);
+        let latest_checkpoint = checkpoint_light_client_client::ServiceState::new(remoting)
+            .get(Order::Reverse, 0, 1)
+            .recv(checkpoint_light_client_address.into())
+            .await
+            .ok()
+            .map(|state| {
+                state
+                    .checkpoints
+                    .last()
+                    .map(|(checkpoint, _)| EthereumSlotNumber(*checkpoint))
+            })
+            .unwrap_or(None);
```

Comment:
Copy-pasta ). Please extract a new function
```suggestion
fn get_last_checkpoint() -> AnyResult<Option<EthereumSlotNumber>> {
        let client = api_provider
            .gclient_client(&suri)?;
        let remoting = GClientRemoting::new(client);
        let state = checkpoint_light_client_client::ServiceState::new(remoting)
            .get(Order::Reverse, 0, 1)
            .recv(checkpoint_light_client_address.into())
            .await?;

         Ok(state
                    .checkpoints
                    .last()
                    .map(|(checkpoint, _)| EthereumSlotNumber(*checkpoint)))
}
```

and use it whenever required.